File: Swift/프로그래머스/2/84512. 모음 사전/모음 사전.swift
import Foundation

func solution(_ word:String) -> Int {
    
    let str_list = ["A", "E", "I", "O", "U"]
    var cnt = 0
    
    func backtracking(_ ans: [String]) -> Bool {
        
        cnt += 1
        
        if ans.joined() == word {
            return true
        }
        
        if ans.count == 5 {
            return false
        }
        
        var temp = ans
        for str in str_list {
            temp.append(str)
            if backtracking(temp) {
                return true
            }
            temp.popLast()
        }
        
        return false
    }
    
    for str in str_list {
        if backtracking([str]) {
            return cnt
        }
    }
        
    return 0
}

File: Python3/프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    
    alpha_list = ["A","E","I","O","U"]
    cnt = 0
    
    def backtracking(ans):
        # print(ans)
        nonlocal cnt
        cnt += 1
        
        if "".join(ans) == word:
            return True
        
        if len(ans) == 5:
            return False
        
        for alpha in alpha_list:
            ans.append(alpha)
            if backtracking(ans):
                return True
            ans.pop()
        
        
    for s in alpha_list:
        if backtracking([s]):
            return cnt
        
    return 0